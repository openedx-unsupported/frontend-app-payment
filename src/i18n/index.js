import { messages as paragonMessages } from '@edx/paragon';
import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';

import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
