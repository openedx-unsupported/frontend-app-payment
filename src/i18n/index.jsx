import arMessages from './messages/ar.json';
import frMessages from './messages/fr.json';
import es419Messages from './messages/es_419.json';
import es419MessagesProtected from '../subscription/i18n-protected-messages/es_419.json';
import zhcnMessages from './messages/zh_CN.json';
import ptMessages from './messages/pt.json';
import itMessages from './messages/it.json';
import ukMessages from './messages/uk.json';
import deMessages from './messages/de.json';
import ruMessages from './messages/ru.json';
import hiMessages from './messages/hi.json';
import frCAMessages from './messages/fr_CA.json';
import deDEMessages from './messages/de_DE.json';
import itITMessages from './messages/it_IT.json';
import ptPTMessages from './messages/pt_PT.json';
import faMessages from './messages/fa.json';
import faIRMessages from './messages/fa_IR.json';
// no need to import en messages-- they are in the defaultMessage field

const messages = {
  ar: arMessages,
  // override the protected translations
  'es-419': { ...es419Messages, ...es419MessagesProtected },
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': deDEMessages,
  'it-it': itITMessages,
  'pt-pt': ptPTMessages,
  fa: faMessages,
  'fa-ir': faIRMessages,
};

export default messages;
