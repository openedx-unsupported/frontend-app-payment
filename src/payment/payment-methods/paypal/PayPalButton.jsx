import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from '@edx/frontend-i18n';

import PayPalLogo from './assets/paypal-logo.png';
import messages from './PayPalButton.messages';

const PayPalButton = ({ intl, working, ...props }) => (
  <button {...props}>
    { working ? <span className="button-spinner-icon text-primary mr-2" /> : null }
    <img
      src={PayPalLogo}
      alt={intl.formatMessage(messages['payment.type.paypal'])}
    />
  </button>
);

PayPalButton.propTypes = {
  intl: intlShape.isRequired,
  working: PropTypes.bool,
};

PayPalButton.defaultProps = {
  working: false,
};

export default injectIntl(PayPalButton);
