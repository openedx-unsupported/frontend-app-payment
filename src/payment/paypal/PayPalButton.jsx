import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { injectIntl, intlShape } from '@edx/frontend-i18n';
import { sendTrackEvent } from '@edx/frontend-analytics';

import PayPalLogo from './assets/paypal-logo.png';
import messages from './PayPalButton.messages';
import { submitPaymentPayPal } from './data/actions';

class PayPalButton extends React.Component {
  handleClick = () => {
    if (this.props.disabled) return;

    // TO DO: after event parity, track data should be
    // sent only if the payment is processed, not on click
    // Check for ApplePay and Free Basket as well
    sendTrackEvent(
      'edx.bi.ecommerce.basket.payment_selected',
      {
        type: 'click',
        category: 'checkout',
        paymentMethod: 'PayPal',
      },
    );
    this.props.submitPaymentPayPal();
  }


  render() {
    const {
      intl, submitting, className, disabled,
    } = this.props;

    return (
      <button
        onClick={this.handleClick}
        className={classNames('payment-method-button', className)}
        disabled={submitting || disabled}
      >
        <img
          src={PayPalLogo}
          alt={intl.formatMessage(messages['payment.type.paypal'])}
        />
      </button>
    );
  }
}

PayPalButton.propTypes = {
  intl: intlShape.isRequired,
  className: PropTypes.string,
  submitting: PropTypes.bool,
  disabled: PropTypes.bool,
  submitPaymentPayPal: PropTypes.func.isRequired,
};

PayPalButton.defaultProps = {
  submitting: false,
  className: undefined,
  disabled: false,
};

const mapStateToProps = state => state.payment.paypal;

export default connect(mapStateToProps, { submitPaymentPayPal })(injectIntl(PayPalButton));
