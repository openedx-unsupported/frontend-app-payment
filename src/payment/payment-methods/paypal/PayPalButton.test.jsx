import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import PayPalButton from './PayPalButton';

describe('OrderDetails', () => {
  it('should render the button by default', () => {
    const component = (
      <IntlProvider locale="en">
        <PayPalButton />
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render the button with a spinner when processing', () => {
    const component = (
      <IntlProvider locale="en">
        <PayPalButton isProcessing />
      </IntlProvider>
    );
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
