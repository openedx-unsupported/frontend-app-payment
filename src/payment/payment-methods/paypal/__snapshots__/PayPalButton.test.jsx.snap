// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetails should render the button by default 1`] = `
<button
  type="button"
>
  <img
    alt="PayPal"
    src="test-file-stub"
  />
</button>
`;

exports[`OrderDetails should render the button with a spinner when processing 1`] = `
<button
  type="button"
>
  <span
    className="button-spinner-icon text-primary mr-2"
  />
  <img
    alt="PayPal"
    src="test-file-stub"
  />
</button>
`;
